cli/lib/otel.ts

```diff
@@ -0,0 +1,56 @@
+import { trace } from '@opentelemetry/api'
+import { CollectorTraceExporter } from '@opentelemetry/exporter-collector'
+import { HttpInstrumentation } from '@opentelemetry/instrumentation-http'
+import { WinstonInstrumentation } from '@opentelemetry/instrumentation-winston'
+import { Resource, envDetector, processDetector } from '@opentelemetry/resources'
+import { NodeSDK } from '@opentelemetry/sdk-node'
+import { ResourceAttributes } from '@opentelemetry/semantic-conventions'
+import { BatchSpanProcessor } from '@opentelemetry/tracing'
+
+export const publishTracer = trace.getTracer('cube-publish')
+
+const sdk = new NodeSDK({
+  // Automatic detection is disabled, see comment below
+  autoDetectResources: false,
+  instrumentations: [
+    new HttpInstrumentation(),
+    new WinstonInstrumentation(),
+  ],
+  resource: new Resource({
+    [ResourceAttributes.SERVICE_NAME]: 'cube-creator-cli',
+  }),
+})
+
+const onError = async (err: Error) => {
+  // Remove signal handler to quit immediately when receiving multiple
+  // SIGINT/SIGTEM
+  process.off('SIGINT', onError)
+  process.off('SIGTERM', onError)
+
+  if (err) {
+    // eslint-disable-next-line no-console
+    console.log(err)
+  }
+  await sdk.shutdown()
+  process.exit(1)
+}
+
+export async function opentelemetry() {
+  try {
+    await sdk.detectResources({ detectors: [envDetector, processDetector] })
+
+    const exporter = new CollectorTraceExporter()
```

Comment:
Since we're not doing that conditionally, it can be simplified by just having this in the SDK creation:
```
const sdk = new NodeSDK({
  traceExporter: new CollectorTraceExporter(),
  // ...
})
```

The `BatchSpanProcessor` is automatically setup by the SDK if you do this.